api/src/main/java/org/openmrs/module/queue/api/impl/QueueEntryServiceImpl.java

```diff
@@ -83,6 +85,17 @@ public QueueEntry saveQueueEntry(QueueEntry queueEntry) {
 				throw new IllegalArgumentException("Patient mismatch - visit.patient does not match patient");
 			}
 		}
+		QueueEntrySearchCriteria searchCriteria = new QueueEntrySearchCriteria();
+		searchCriteria.setPatient(queueEntry.getPatient());
+		searchCriteria.setQueues(Collections.singletonList(queueEntry.getQueue()));
+		List<QueueEntry> queueEntries = getQueueEntries(searchCriteria);
+		for (QueueEntry qe : queueEntries) {
+			if (!qe.equals(queueEntry)) {
+				if (QueueUtils.datesOverlap(qe.getStartedAt(), qe.getEndedAt(), qe.getStartedAt(), qe.getEndedAt())) {
+					throw new DuplicateQueueEntryException("queue.entry.duplicate.patient");
```

Comment:
@cioan - note, I removed the translation of the exception altogether.  This can be handled downstream in the view layer.